mlir/lib/Pass/PassCrashRecovery.cpp

```diff
@@ -283,19 +289,7 @@ void PassCrashReproducerGenerator::finalize(Operation *rootOp,
   impl->runningPasses.clear();
 }
 
-void PassCrashReproducerGenerator::prepareReproducerFor(Pass *pass,
-                                                        Operation *op) {
-  // If not tracking local reproducers, we simply remember that this pass is
-  // running.
-  impl->runningPasses.insert(std::make_pair(pass, op));
-  if (!impl->localReproducer)
-    return;
-
-  // Disable the current pass recovery context, if there is one. This may happen
-  // in the case of dynamic pass pipelines.
-  if (!impl->activeContexts.empty())
-    impl->activeContexts.back()->disable();
-
+static std::string produceTextualPipelineWithScope(Pass *pass, Operation *&op) {
```

Comment:
Why is the `op` pointer passed by reference?

Reply:
Because the while-loop in the function updates it. The refactor was breaking before I passed by reference until I figured this out. I will see if I can drop refactoring out the produceTextualPipelineWithScope function however.

Reply:
OK I see. If you have to keep this, please add a comment before the function describing the update.